src/utils.py

```diff
@@ -1,24 +1,68 @@
-import logging
 import os
 import random
 import sys
 import time
-
+import queue
+import threading
 from selenium import webdriver
 from loguru import logger
 
-from app_config import MINIMUM_LOG_LEVEL
-
-log_file = "app_log.log"
-
-
-if MINIMUM_LOG_LEVEL in ["DEBUG", "TRACE", "INFO", "WARNING", "ERROR", "CRITICAL"]:
-    logger.remove()
-    logger.add(sys.stderr, level=MINIMUM_LOG_LEVEL)
-else:
-    logger.warning(f"Invalid log level: {MINIMUM_LOG_LEVEL}. Defaulting to DEBUG.")
-    logger.remove()
-    logger.add(sys.stderr, level="DEBUG")
+from app_config import MINIMUM_LOG_LEVEL, LOG_TO_FILE, LOG_TO_CONSOLE
+
+log_file = "log/app.log"
+
+# Ensure the log directory exists
+os.makedirs(os.path.dirname(log_file), exist_ok=True)
+
+# Remove default logger
+logger.remove()
+
+# Configure Loguru logger
+config = {
+    "handlers": []
+}
+
+# Add file logger if LOG_TO_FILE is True
+if LOG_TO_FILE:
```

Comment:
can you use simple logging handler? it's not wise to reinvent the well.

Python do profile FileHandler, BaseRotatingHandler and other things.

Reply:
and remove multithreading, as it is adding complexity which is completely unnecessary.